janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java

```diff
@@ -571,10 +568,17 @@ private void tokenize(SearchParams params, final Mapping mapping, final LuceneCu
                 params.addQuery(new PrefixQuery(term), BooleanClause.Occur.MUST);
             } else throw new IllegalArgumentException("LuceneIndex does not support this predicate with 1 token : " + janusgraphPredicate);
         } else {
-            // at the moment, this is only walked for EQUAL and Text.CONTAINS (String and Text mappings)
+            // at the moment, this is only walked for EQUAL, NOT_EQUAL and Text.CONTAINS (String and Text mappings)
             final BooleanQuery.Builder q = new BooleanQuery.Builder();
-            for (final String term : terms) {
-                q.add(new TermQuery(new Term(key, term)), BooleanClause.Occur.MUST);
+            BooleanClause.Occur occur;
+            if (janusgraphPredicate == Cmp.NOT_EQUAL) {
+                q.add(new MatchAllDocsQuery(), BooleanClause.Occur.MUST);
+                q.add(new TermQuery(new Term(key, value)), BooleanClause.Occur.MUST_NOT);
```

Comment:
Why we use `value` when we have multiple terms and `terms.get(0)` when we have a single term for `NOT_EQUAL`?